In [1]:
import pandas as pd

The useful files are:
<p>
"data/course_classroom.csv","data/course_match.csv","data/course_match_student_match.csv",
"data/course_questionresponse.csv","data/course_questionresponse_new_response.csv",
"data/course_questionresponse_new_subresponse.csv","data/course_subquestionresponse.csv",
"data/user_app_address.csv","data/user_app_customuser.csv","data/user_messages_message.csv".
</p>

In [6]:
files=["data/course_match.csv","data/course_questionresponse.csv","data/course_questionresponse_new_response.csv","data/course_questionresponse_new_subresponse.csv","data/course_subquestionresponse.csv","data/user_app_address.csv","data/user_app_customuser.csv","data/user_messages_message.csv"]

In [7]:
def printFile(filePath):
    with open(filePath,'r') as f:
        #data = pd.read_csv(f,delim_whitespace=True,header=None,error_bad_lines=False)
        data = pd.read_csv(f)
        print "=================",filePath,"================="
        #print "summary:\n",data.describe(),"\n\nhead:"
        print data.head(10),'\n'
        #return data

In [8]:
for fi in files:
    printFile(fi)

================= data/course_match.csv =================
      id  student_id  classroom_id  \
0  97395      123884        8253.0   
1  99041      125418        7989.0   
2  12327       17867        1998.0   
3  12632       18857        1911.0   
4  45091       60318        4072.0   
5     50         154          69.0   
6     44         148          69.0   
7     32         134          69.0   
8     34         138          69.0   
9  12180       18321        1965.0   

                                               intro new_match  age  \
0                Australia beacause they have coalas         f   13   
1       Paris, France because it's the city of love.         f   13   
2  We are a free country.\nWe drink water.\nWe bo...         t   13   
3                                                NaN         t   15   
4                                Watch a hockey game         t   10   
5                                                NaN         f   12   
6                         

/Users/Wenbo/anaconda/lib/python2.7/site-packages/IPython/core/interactiveshell.py:2821: DtypeWarning: Columns (2) have mixed types. Specify dtype option on import or set low_memory=False.
  if self.run_code(code, result):


================= data/user_app_address.csv =================
   id street_line1 street_line2 zipcode       city     state          country  \
0   2          NaN          NaN     NaN         SF       CA    United States    
1   3          NaN          NaN     NaN    Watford  England   United Kingdom    
2   4          NaN          NaN     NaN     Austin       TX    United States    
3   5          NaN          NaN     NaN     Austin       TX    United States    
4   6          NaN          NaN     NaN         CF       NaN       Argentina    
5   7          NaN          NaN     NaN        NaN       NaN             None   
6   8          NaN          NaN     NaN        NaN       NaN             None   
7   9          NaN          NaN     NaN        NaN       NaN             None   
8  39          NaN          NaN     NaN  Wadsworth        IL    United States   
9  11          NaN          NaN     NaN        NaN       NaN             None   

             formatted_address        lat     

/Users/Wenbo/anaconda/lib/python2.7/site-packages/IPython/core/interactiveshell.py:2821: DtypeWarning: Columns (13) have mixed types. Specify dtype option on import or set low_memory=False.
  if self.run_code(code, result):


#### First, how to define active and inactive stuent?
<p>I would like to use messages tables and response tables. The word count ,the frequency, the amount of total response, etc data will be used to identify whether a student is activie or not.</p>
#### Second, what I should use for classification?
<p>their profile data. Because Joe said they want to classify the student before the matching/class </p>

### data/course_match.csv
Select most of columns and see if there is a trends.

In [2]:
data_1=pd.read_csv("data/course_match.csv",usecols=['student_id','intro','new_match','age','course_start_date','target_start_date','skill','state','country','activity','points','pledge'])

#### ? Don't know pledge. Also the id in original csv file.

In [3]:
data_1.head()

student_id                                              intro new_match  \
0      123884                Australia beacause they have coalas         f   
1      125418       Paris, France because it's the city of love.         f   
2       17867  We are a free country.\nWe drink water.\nWe bo...         t   
3       18857                                                NaN         t   
4       60318                                Watch a hockey game         t   

   age course_start_date target_start_date         skill state        country  \
0   13        2017-01-09        2017-01-09  Intermediate   NaN        Denmark   
1   13        2017-01-09        2017-01-09  Intermediate    FL  United States   
2   13        2015-05-25        2015-08-31      Beginner    AL  United States   
3   15               NaN        2015-08-31      Beginner    TX  United States   
4   10        2016-02-01        2016-01-18      Beginner    OR  United States   

                        activity  points pledge  
0  2017-01-26 09:03:08.858457+00     300      t  
1  2017-01-25 19:19:34.060776+00     300      t  
2  2015-08-30 19:15:09.928069+00     150      f  
3  2015-08-30 19:14:26.770596+00       0      f  
4   2016-01-29 20:15:01.09436+00     100      f

In [4]:
data_1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 91259 entries, 0 to 91258
Data columns (total 12 columns):
student_id           91259 non-null int64
intro                74651 non-null object
new_match            91259 non-null object
age                  91259 non-null int64
course_start_date    76007 non-null object
target_start_date    91259 non-null object
skill                91259 non-null object
state                89681 non-null object
country              91248 non-null object
activity             91259 non-null object
points               91259 non-null int64
pledge               91259 non-null object
dtypes: int64(3), object(9)
memory usage: 8.4+ MB


In [5]:
data_1.describe()

student_id           age        points
count   91259.000000  91259.000000  91259.000000
mean    71835.266878     12.593476    203.267075
std     36299.815653     26.125831    127.896151
min        88.000000    -71.000000      0.000000
25%     38751.500000     11.000000    100.000000
50%     75364.000000     12.000000    200.000000
75%    102613.500000     14.000000    300.000000
max    130738.000000   1996.000000    450.000000

#### Hypothesis
There is 91259 records, 74651 of them have introduction, 76007 have course_start_date, 89681 have state. SO:
<li>Not all records has "state" maybe because states are not available for some foreign contry.</li>
<li>? need to know why there are some records with no course_start_date.</li>
<li> some students may have no introduction.</li>
And from the statistic data:
<li> Age data has some problems: the min value should not be negative; the max value should not be 1996, which could be born year.</li>

In [ ]:
data_1.pop()

In [9]:
data_2 = pd.read_csv("data/course_questionresponse.csv")
data_2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 445977 entries, 0 to 445976
Data columns (total 7 columns):
id             445977 non-null int64
created        445977 non-null object
modified       445977 non-null object
match_id       445977 non-null int64
question_id    255484 non-null float64
text           445973 non-null object
unit_id        255484 non-null float64
dtypes: float64(2), int64(2), object(3)
memory usage: 23.8+ MB


In [10]:
data_2.describe()

id       match_id    question_id        unit_id
count  445977.000000  445977.000000  255484.000000  255484.000000
mean   293172.854874   50610.124193     415.205876      32.390005
std    157232.444675   28513.655944     258.006175      21.902441
min         2.000000      32.000000       2.000000       1.000000
25%    163848.000000   25333.000000     131.000000       4.000000
50%    300091.000000   49898.000000     393.000000      40.000000
75%    428269.000000   75802.000000     701.000000      43.000000
max    556424.000000  104177.000000     732.000000      96.000000

#### Question
question response only has match_id, what does id means? How can I know the response is made by whom? So I must use match table to find match id and student id? 